crates/turbopack-css/src/chunk/mod.rs

```diff
@@ -65,21 +63,65 @@ impl CssChunk {
 
         let this = self.await?;
 
+        let mut code = CodeBuilder::default();
         let mut body = CodeBuilder::default();
-        let mut external_imports = IndexSet::new();
-        for css_item in this.content.await?.chunk_items.iter() {
+        let mut seen = HashSet::new();
+        for css_item in &this.content.strongly_consistent().await?.chunk_items {
             // TODO(WEB-1261)
-            for external_import in expand_imports(&mut body, *css_item).await? {
-                external_imports.insert(external_import.await?.to_owned());
+            let id = &*css_item.id().await?;
+            let content = &*css_item.content().await?;
+
+            let attr = match content.import_context {
+                Some(i) => Some((*i.await?).clone()),
+                None => None,
+            };
+
+            let key = (id.clone(), attr.clone());
+            if seen.contains(&key) {
+                continue;
             }
-        }
+            seen.insert(key);
 
-        let mut code = CodeBuilder::default();
-        for external_import in external_imports {
-            writeln!(code, "@import {};", stringify_js(&external_import))?;
+            let content = &css_item.content().await?;
+            for import in &content.imports {
+                if let CssImport::External(external_import) = import {
+                    writeln!(code, "@import {};", stringify_js(&external_import.await?))?;
```

Comment:
`@import` can only appear at the start of a css file. You need to collect all external css first and make sure they are emitted first

Reply:
This should do that. The imports are written directly to `code`, while file contents are added to `body`, which is added to `code` only after the items have been processed.